### PR TITLE
Cu 2cyqzuh create submission microservice

### PR DESCRIPTION
**Pull request by 0c370t**

### CODE REVIEW

**Comment by 0c370t** [comment]
Task linked: [CU-2cyqzuh Create Submission Microservice](https://app.clickup.com/t/2cyqzuh)

File: microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts [hunk 1]
```diff
@@ -1,33 +1,105 @@
...
+
+        const scrim = await this.scrimCrudService.getScrimBySubmissionId(payload.submissionId);
+        if (!scrim) {
+            this.logger.warn(`Scrim not found for submission ${payload.submissionId}`);
+            return;
+        }
+        await this.scrimService.completeScrim(scrim.id);
+    };
+
+    onSubmissionRejected = async (d: EventResponse<EventTopic.SubmissionRejected | EventTopic>): Promise<void> => {
```
Comment (gankoji) [inline microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts:94]: A lot of repeated code here in a pattern. Seems odd that we'd switch on the event type, to call this boilerplate functions, that effectively just move a state machine forward once errors are handled. There's a cleaner way to implement state machines for sure, not necessary for this effort though.

File: microservices/submission-service/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+I would like to coerce a build in this project.
```
Comment (gankoji) [inline microservices/submission-service/README.md:1]: Might as well remove this now that we're wrapping up.

**Review by gankoji — APPROVED** [review]

File: clients/web/src/lib/api/queries/Submission.store.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    `;
+
+    constructor(submissionId: string) {
+        super();
+        this.vars = {submissionId};
+        this.subscriptionVariables = {submissionId};
+    }
+
+    protected handleGqlMessage = (message: OperationResult<SubmissionSubscriptionValue, SubmissionStoreSubscriptionVariables>): void => {
+        if (message.data?.submission) {
```
[the remark this answers is not in the document]
Comment (0c370t) [reply clients/web/src/lib/api/queries/Submission.store.ts:158]: Not needed because we are reassigning the entire object

**Pull request merged**